### ISSUE TITLE
[question] consume json from Ushahidi

### ISSUE THREAD

**Issue by Lunkster** [calvinmetcalf/leaflet-ajax#20]
Hi all
I try to consume locations from a Ushahidi website which deliver a kind of "dirty" json but it passas as valid in online json validators.

I get an error at leaflet.ajax.js, row 547: SyntaxError JSON parse: unexpected end of data.

http://kajakplatser.se/json

Is there a solution for this or is Ushahidi v2.7 json inedible? :-)

Cheers
Henrik

**Comment by calvinmetcalf** [comment]
could you throw together a jsfiddle so I can see exactly what your doing?

**Comment by Lunkster** [comment]
The code I use is as follows, under test and somewhat dirty of course. Sent
from phone and not fiddled.

Cheers

<HTML>

<HEAD>
<title>Kajakplatser</title>
<meta charset="utf-8">

<link rel="stylesheet" href="assets/map/leaflet/leaflet.css" />
<link rel="stylesheet" href="assets/css/map.css" />

<style type="text/css">
body {
padding: 0;
margin: 0;
}
html, body, #map {
height: 100%;
}
a {
color: gray;
}
#overlay {
z-index: 10;
position: absolute;
top: 12px;
left: 60px;
width: 50%;
min-width: 230px;
padding: 5px;
}
#overlay.hidden {
width: 19px;
min-width: 0px;
}
#overlay,
.leaflet-control-zoom,
.leaflet-control-layers {
font: 12px/1.5 "Helvetica Neue", Arial, Helvetica, sans-serif;
color: rgb(255,255,255);
box-shadow: none;
-moz-border-radius: 7px;
-webkit-border-radius: 7px;
border-radius: 7px;
background: rgba(0,0,0,0.7);
}
.content {
margin: 10px;
}
.hidden .content {
display: none;
}
#toggle {
z-index: 100;
-moz-border-radius: 4px;
-webkit-border-radius: 4px;
border-radius: 4px;
width: 19px;
height: 19px;
background-color: rgba(255, 255, 255, 0.75);
}
#toggle:hover {
cursor: pointer;
background: white;
}
</style>

<script type="text/javascript" src="leaflet/leaflet.js"></script>

<script type="text/javascript" src="src/lvector.js"></script>

<script type="text/javascript" src="src/layer/Layer.js"></script>

<script src="assets/map/leaflet-hash.js"></script>

<script src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js
"></script>
<script type="text/javascript" src="
http://maps.stamen.com/js/tile.stamen.js?v1.2.1"></script>
<script src="assets/OSMBuildings-Leaflet.js"></script>
<script src="http://maps.google.com/maps/api/js?v=3&sensor=false"></script>
<script src="leaflet-plugins-master/layer/tile/Google.js"></script>
<script type="text/javascript"
src="leaflet-ajax/dist/leaflet.ajax.js"></script>
<script src="leaflet-ajax/example/spin.js"></script>
<script src="leaflet-ajax/example/leaflet.spin.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
</HEAD>
<BODY>
<div id="overlay">
<div id="toggle"></div>
<div class="content">
<h3>Kajakplatser</h3>
<p>
Dessa kajakplatser hämtas från kajakplatser.se som drivs mha Ushahidi <br/>
</div>
</div>
<div id="geolocate">
         <a href="#" title="Hitta mig"><img
src="assets/image/location.png"/></a>
</div>
<div id="map"></div>
<script type="text/javascript" charset="utf-8">
$("#overlay").toggleClass("hidden");
$('#toggle').click(function () {
$("#overlay").toggleClass("hidden");
});
</script>

<script type="text/javascript">
// Open Street Map som lager
var osm = new L.TileLayer("http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
{
attribution: "Map data &copy; 2013 OpenStreetMap contributors"
});
// Open Cycle Map som lager
var ocm = new L.TileLayer("http://{s}.
tile.opencyclemap.org/cycle/{z}/{x}/{y}.png", {
attribution: "Map data &copy; 2013 OpenStreetMap contributors"
});
// Mapquest som lager
var mapquest = new L.TileLayer("
http://otile1.mqcdn.com/tiles/1.0.0/osm/{z}/{x}/{y}.jpg", {
attribution: 'Map data &copy; 2013 OpenStreetMap contributors, Tiles &copy;
<a href="http://www.mapquest.com/" target="_blank">MapQuest</a> <img src="
http://developer.mapquest.com/content/osm/mq_logo.png" />'
});
// Stamen watercolor som lager
    var stamenwatercolor = new L.StamenTileLayer("watercolor", {
attribution: 'Map data &copy; 2013 OpenStreetMap contributors, Tiles &copy;
<a href="http://maps.stamen.com/" target="_blank">Stamen maps</a>'
});
// Stamen toner-lite som lager
    var stamentonerlite = new L.StamenTileLayer("toner-lite", {
attribution: 'Map data &copy; 2013 OpenStreetMap contributors, Tiles &copy;
<a href="http://maps.stamen.com/" target="_blank">Stamen maps</a>'
});
// Hydda från kodapan
    var hydda = new L.TileLayer("
http://keb.kodapan.se/hydda/full/{z}/{x}/{y}.png", {
maxZoom: 18,
attribution: "Map data &copy; 2011 OpenStreetMap contributors, layout
kodapan.se"
});
// Google maps
var ggl = new L.Google();
var map = L.map('map', {
layers: [hydda],
zoomControl: true
}).setView([57.85, 14.10], 12);
var hash = new L.Hash();
hash.init(map);
var geolocate = document.getElementById('geolocate');
geolocate.onclick = function() {
function onLocationFound(e) {
L.marker(e.latlng).addTo(map)
.bindPopup("H&auml;r &auml;r du!").openPopup();
}
function onLocationError(e) {
alert(e.message);
}
map.on('locationfound', onLocationFound);
map.on('locationerror', onLocationError);
map.locate({setView: true});
}
var osmb = new OSMBuildings(map).loadData();
function popUp(f,l){
    var out = [];
    if (f.properties){
        for(key in f.properties){
            out.push(key+": "+f.properties[key]);
        }
        l.bindPopup(out.join("<br />"));
    }
}
var kajakplatser = new
L.GeoJSON.AJAX("http://kajakplatser.se/json",{onEachFeature:popUp,
dataType:"json"}).addTo(map);
var baseMaps = {
"Stamen Watercolor": stamenwatercolor,
"Stamen Toner Lite": stamentonerlite,
"Open Street Map": osm,
"Open Cycle Map": ocm,
"Mapquest": mapquest,
"Hydda komplett": hydda,
"Google": ggl,
};
var overlayMaps = {
"Kajakplatser": kajakplatser,
"Byggnader 3D": osmb,
};
// Lägg till en lagerkontroll
L.control.layers(baseMaps, overlayMaps).addTo(map);
L.control.scale().addTo(map);
</script>

</BODY>
</HTML>
Den 22 okt 2013 01:27 skrev "Calvin Metcalf" notifications@github.com:

**Comment by Lunkster** [comment]
Oh... sorry for the total lack of formatting...
Den 22 okt 2013 01:27 skrev "Calvin Metcalf" notifications@github.com:

**Comment by calvinmetcalf** [comment]
`XMLHttpRequest cannot load http://kajakplatser.se/json. Origin http://calvinmetcalf.github.io is not allowed by Access-Control-Allow-Origin.` http://kajakplatser.se/json doesn't have cors turned on

**Comment by calvinmetcalf** [comment]
as of 01d15576344dd391b8aba2222d7dd7f0ac76ae0d it now gives better errors for cross origin issues

**Comment by Lunkster** [comment]
Thanks for the info. It seems that the jsonp-url should be as follows:
http://kajakplatser.se/api?task=incidents&resp=jsonp&callback=callback

When I try that I don't get any error messages, but no incidents show up. Any ideas?
Cheers
Henrik

**Comment by calvinmetcalf** [comment]
the data at that endpoint isn't actually geojson, you can use the jsonp function to download and then manually add it to the map if you want

``` javascript
L.Util.jsonp('http://kajakplatser.se/api?task=incidents&resp=jsonp').then(function(data){
//do stuff
});
```

**Comment by Lunkster** [comment]
ah... thanks for your time. :-)

**Comment by calvinmetcalf** [comment]
this isn't open layers

On Mon, Jul 21, 2014 at 4:50 AM, neogeomat notifications@github.com wrote:

> How can i do the jsonp request with OpenLayers? i tried roadreport =
> OpenLayers.Request.GET({url:"
> http://kathmandulivinglabs/roadreporter/api/?task=incidents&resp=jsonp&callback=callback
> "}) and it gives net::ERR_NAME_NOT_RESOLVED error.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/calvinmetcalf/leaflet-ajax/issues/20#issuecomment-49583275
> .

## 

-Calvin W. Metcalf

**Comment by calvinmetcalf** [comment]
you might try http://gis.stackexchage.com

On Mon, Jul 21, 2014 at 5:17 AM, Calvin Metcalf calvin.metcalf@gmail.com
wrote:

> this isn't open layers
> 
> On Mon, Jul 21, 2014 at 4:50 AM, neogeomat notifications@github.com
> wrote:
> 
> > How can i do the jsonp request with OpenLayers? i tried roadreport =
> > OpenLayers.Request.GET({url:"
> > http://kathmandulivinglabs/roadreporter/api/?task=incidents&resp=jsonp&callback=callback
> > "}) and it gives net::ERR_NAME_NOT_RESOLVED error.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/calvinmetcalf/leaflet-ajax/issues/20#issuecomment-49583275
> > .
> 
> ## 
> 
> -Calvin W. Metcalf

## 

-Calvin W. Metcalf
